src/components/ha-selector/ha-selector-device.ts

```diff
@@ -38,6 +38,12 @@ export class HaDeviceSelector extends LitElement {
       .value=${this.value}
       .label=${this.label}
       .deviceFilter=${(device) => this._filterDevices(device)}
+      .includeDeviceClasses=${this.selector.device.entity?.device_class
+        ? [this.selector.device.entity?.device_class]
+        : undefined}
+      .includeDomains=${this.selector.device.entity?.domain
+        ? [this.selector.device.entity?.domain]
```

Comment:
It's weird you need to write `entity?` instead of `entity` because you're inside a ternary statement that you can only be if `entity` is defined.

Reply:
copy paste,... :-/